services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/document/EnhancedDocument.java

```diff
@@ -0,0 +1,603 @@
+     * <li>map (with string key to value of any of the types on this list)</li>
+     * <li>set (of any of the types on this list)</li>
+     * </ul>
+     *
+     * @param attributeName Name of the attribute.
+     * @return value of the specified attribute in the current document as an object; or null if the attribute either doesn't
+     * exist or the attribute value is null.
+     */
+    Object get(String attributeName);
+
+    /**
+     * Gets the EnhancedTyoe for the specified attribute key *
+     *
+     * @param attributeName Name of the attribute.
+     * @return type of the specified attribute in the current item; or null if the attribute either doesn't exist or the attribute
+     * value is null.
+     */
+    EnhancedType<?> getTypeOf(String attributeName);
+    
+    /**
+     * Iterable for all the attributes in the enhancedDocument
+     *
+     * @return all attributes of the current item.
+     */
+    Iterable<Map.Entry<String, Object>> attributes();
+
+    /**
+     * Gets the current enhancedDocument as Map.
+     *
+     * @return attributes of the current document as a map.
+     */
+    Map<String, Object> asMap();
+
+    /**
+     * Returns the number of attributes of this item.
+     */
+    int numberOfAttributes();
+
+    /**
+     * *
+     * @return document as a JSON string.  
+     * Note all binary data will become base-64 encoded in the resultant string.
+     */
+    String toJSON();
+
+    /**
+     * Utility method to decode the designated binary attributes from base-64 encoding; converting binary lists into binary sets
+     *
+     * @param binaryAttrNames names of binary attributes or binary set attributes currently base-64 encoded (typically when *
+     *                        converted from a JSON string.)
+     * @return decode the designated binary attributes from base-64 encoding
+     */
+    EnhancedDocument base64Decode(String... binaryAttrNames);
+
+    /**
+     * Gets the entire enhanced document as
+     *
+     * @return document as a pretty JSON string. Note all binary data will become base-64 encoded in the resultant string
+     */
+    String toJSONPretty();
+
+
+    @NotThreadSafe
+    interface Builder {
+        /**
+         * Adds key attribute with the given value to the Document. An attribute value can be a
+         * <ul>
+         *  <li>Number</li>
+         *  <li>String</li>
+         *  <li>binary (ie byte array or byte buffer)</li>
+         *  <li>boolean</li>
+         *  <li>null</li>
+         *  <li>list (of any of the types on this list)</li>
+         *  <li>map (with string key to value of any of the types on this list)</li>
+         *  <li>set (of any of the types on this list)</li>
+         * </ul>
+         *
+         * @param attributeName Name of the attribute that needs to be added in the Document builder.
+         * @param val           Value of the specified attribute
+         * @return Builder instance to construct a {@link EnhancedDocument}
```

Comment:
Just wondering: Do we want to align the indents for the description after the @return? Here and below